packages/airnode-protocol/test/utils/SelfServeRrpBeaconServerWhitelister.sol.js

```diff
@@ -0,0 +1,363 @@
+beforeEach(async () => {
+  const accounts = await hre.ethers.getSigners();
+  roles = {
+    deployer: accounts[0],
+    manager: accounts[1],
+    sponsor: accounts[6],
+    updateRequester: accounts[7],
+    beaconReader: accounts[8],
+    anotherBeaconReader: accounts[9],
+    randomPerson: accounts[10],
+  };
+  const accessControlRegistryFactory = await hre.ethers.getContractFactory('AccessControlRegistry', roles.deployer);
+  accessControlRegistry = await accessControlRegistryFactory.deploy();
+  const airnodeRrpFactory = await hre.ethers.getContractFactory('AirnodeRrp', roles.deployer);
+  airnodeRrp = await airnodeRrpFactory.deploy();
+  const rrpBeaconServerFactory = await hre.ethers.getContractFactory('RrpBeaconServer', roles.deployer);
+  rrpBeaconServer = await rrpBeaconServerFactory.deploy(
+    accessControlRegistry.address,
+    rrpBeaconServerAdminRoleDescription,
+    roles.manager.address,
+    airnodeRrp.address
+  );
+  const selfServeRrpBeaconServerWhitelisterFactory = await hre.ethers.getContractFactory(
+    'SelfServeRrpBeaconServerWhitelister',
+    roles.manager
+  );
+
+  ({ airnodeAddress, airnodeMnemonic, airnodeXpub } = testUtils.generateRandomAirnodeWallet());
+  airnodeWallet = hre.ethers.Wallet.fromMnemonic(airnodeMnemonic, "m/44'/60'/0'/0/0");
+  sponsorWalletAddress = testUtils.deriveSponsorWalletAddress(airnodeXpub, roles.sponsor.address);
+  await roles.deployer.sendTransaction({
+    to: sponsorWalletAddress,
+    value: hre.ethers.utils.parseEther('1'),
+  });
+  sponsorWallet = testUtils.deriveSponsorWallet(airnodeMnemonic, roles.sponsor.address).connect(hre.ethers.provider);
+  endpointId = testUtils.generateRandomBytes32();
+  endpointId2 = testUtils.generateRandomBytes32();
+  templateParameters = testUtils.generateRandomBytes();
+  await airnodeRrp.connect(roles.randomPerson).createTemplate(airnodeAddress, endpointId, templateParameters);
+  await airnodeRrp.connect(roles.randomPerson).createTemplate(airnodeAddress, endpointId2, templateParameters);
+  templateId = hre.ethers.utils.keccak256(
+    hre.ethers.utils.solidityPack(['address', 'bytes32', 'bytes'], [airnodeAddress, endpointId, templateParameters])
+  );
+  templateId2 = hre.ethers.utils.keccak256(
+    hre.ethers.utils.solidityPack(['address', 'bytes32', 'bytes'], [airnodeAddress, endpointId2, templateParameters])
+  );
+
+  beaconParameters = testUtils.generateRandomBytes();
+  beaconId = hre.ethers.utils.keccak256(
+    hre.ethers.utils.solidityPack(['bytes32', 'bytes'], [templateId, beaconParameters])
+  );
+  beaconId2 = hre.ethers.utils.keccak256(
+    hre.ethers.utils.solidityPack(['bytes32', 'bytes'], [templateId2, beaconParameters])
+  );
+
+  await airnodeRrp.connect(roles.sponsor).setSponsorshipStatus(rrpBeaconServer.address, true);
+  await rrpBeaconServer.connect(roles.sponsor).setUpdatePermissionStatus(roles.updateRequester.address, true);
+
+  // Compute the expected request ID
+  const requestId = hre.ethers.utils.keccak256(
```

Comment:
Nit: Shouldn't this be moved to some utils file?

Reply:
this one definently can be moved to utils